lib/marqeta/api_caller.rb

```diff
@@ -10,17 +10,27 @@ def initialize(endpoint, params = {})
 
     def get
       logger.info("GET: #{endpoint}")
-      response = resource.get
-      logger.info("Response: #{response}")
-      JSON.parse(response)
+      begin
+        response = resource.get
+        handle_successful_response response
+      rescue RestClient::ExceptionWithResponse => e
+        handle_exception_with_response e
+      rescue *HttpError::ERROR_LIST => e
+        handle_http_error e
```

Comment:
Not critical for now, but we could have a fallthrough here for `rescue` (i.e. `rescue StandardError`) to catch anything which isn't caught by the above, and perhaps raise Marqueta::UnknownError.

Reply:
Out of curiosity why would we want to do this vs just having the actual error raised?